packages/create-jest/package.json

```diff
@@ -0,0 +1,31 @@
+{
+  "name": "create-jest",
+  "description": "Create a new Jest project",
+  "version": "29.6.4",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/jestjs/jest.git",
+    "directory": "packages/create-jest"
+  },
+  "license": "MIT",
+  "bin": "./bin/create-jest.js",
+  "main": "./build/index.js",
+  "types": "./build/index.d.ts",
+  "exports": {
+    ".": {
+      "types": "./build/index.d.ts",
+      "default": "./build/index.js"
+    },
+    "./package.json": "./package.json",
+    "./bin/create-jest": "./bin/create-jest.js"
+  },
+  "dependencies": {
+    "jest-cli": "workspace:^"
```

Comment:
This means that the whole Jest with all its dependencies will be pulled? Or? I would expect the `create-` package to be a lightweight script. Did I miss something?

Reply:
Perhaps it makes sense to move the whole logic to the new package and deprecate the `--init` flag in Jest 30?

Reply:
> This means that the whole Jest with all its dependencies will be pulled? Or? I would expect the create- package to be a lightweight script. Did I miss something?

I was thinking about standardised global wrapper for the initialisation (adapter for `npx jest --init`). For this purpose whole `jest-cli` is not necessary indeed.

> Perhaps it makes sense to move the whole logic to the new package and deprecate the --init flag in Jest 30?

Sounds fine, or maybe we can reverse the dependencies and use `create-jest` inside `jest-cli` to keep the API the same

Reply:
In this case `create-jest` would get shipped with `jest`. I was thinking a new package like `@jest/init` could be introduced, but that requires major release as well. Other question, is there a need to have the `--init` command if `create-jest` gets introduced? By the way, moving the whole `--init` logic to `create-jest` would make `jest` leaner (e.g. only `create-jest` would ship `prompts` as dependency, if I didn’t miss something).

Reply:
One more idea, I think for now it would be fine to copy the code to `create-jest` and in `jest-cli` it could be marked with `// TODO Remove in Jest 30` comment.

Reply:
> In this case create-jest would get shipped with jest. I was thinking a new package like @jest/init could be introduced, but that requires major release as well. Other question, is there a need to have the --init command if create-jest gets introduced? By the way, moving the whole --init logic to create-jest would make jest leaner (e.g. only create-jest would ship prompts as dependency, if I didn’t miss something).

I think you're right, in general moving `init` outside of `jest` deps is definitely a right way (though we could invite someone to get a fresh look), because you don't need `init`-related packages during the project lifetime, you need it only during project setup, so it's nice if developers don't have to install unrelated packages each time.

>  in jest-cli it could be marked with // TODO Remove in Jest 30 comment.

Is it a part of the release process to review such comments during the release of major? I'm worried that it may get lost

Reply:
Exactly, `--init` is not much useful later. That’s why it felt like shaving it off is right thing.

Easy to make the `// TODO` more visible: a PR which is removing that code could be opened and I can add it to Jest 30 milestone.

Reply:
> Easy to make the // TODO more visible: a PR which is removing that code could be opened and I can add it to Jest 30 milestone.

Ok, sounds good, I'll add necessary changes here and will prepare an additional PR for major

Reply:
Lots of comments here, and we might all already be in agreement 😀 But just to make sure I'm on the same page as y'all

- This PR will move the entirety of the `--init` command implementation into this new package
- `jest-cli` will then have a dependency on `create-jest`, and delegate to it when `jest --init` is called
- A separate PR will be prepared for v30 (I'll add it to the milestone so we don't forget) which deprecates `--init` and removes the dependency

Sounds about right? If so, I'm all in favour 👍

Reply:
> * `jest-cli` will then have a dependency on `create-jest`, and delegate to it when `jest --init` is called

Could be like this as well, but initially I was thinking:

* `jest-cli` for now stays as it is (gets removed in Jest 30)
* the logic is simply copied `create-jest` and adjusted to work like this: `npx create-jest@latest [rootPath]`.

The `--init` flag is a boolean, but it feels like `create-jest` should take a path arg. Ah.. in a way `jest --init` and `npx create-jest@latest ./` are equivalent, so `create-jest` package can be reused via its programmatic entry (e.g. some `runCreate(rootPath: string): Promise<void>` function, perhaps?).

Reply:
Any reason to duplicate the code. The programmatic API you mention makes sense - why can't `jest --init` use it?

Reply:
That's right. It can reused, I agree.

Reply:
To summarize: in addition to this
> 
> * This PR will move the entirety of the `--init` command implementation into this new package
> * `jest-cli` will then have a dependency on `create-jest`, and delegate to it when `jest --init` is called
> * A separate PR will be prepared for v30 (I'll add it to the milestone so we don't forget) which deprecates `--init` and removes the dependency

we are also adding `[rootPath]` to JS and CLI API, right?

Reply:
> we are also adding `[rootPath]` to JS and CLI API, right?

How do you feel? I find it useful, but can be implemented later.

Reply:
Yeah, I think that's fine. `--init` will default to `process.cwd()`, but I do think it makes sense as part of the API

Reply:
> How do you feel? I find it useful, but can be implemented later.

I also find it useful, makes sense to do it here I think, it's not that hard